JumpRoyale/src/Jumper.cs

```diff
@@ -53,13 +56,37 @@ public partial class Jumper : CharacterBody2D
     public void Init(int x, int y, [NotNull] PlayerData playerData)
     {
         PlayerData = playerData;
-
         Position = new Vector2(x, y);
         Name = PlayerData.Name;
+    }
+
+    public override void _Ready()
+    {
+        _animatedSprite2D = GetNode<AnimatedSprite2D>(SpriteNodeName);
```

Comment:
GDScript has `@onready` for this ([reference](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#onready-annotation)), so I was hoping to be able to point you to the equivalent in C#, but apparently [it still doesn't exist](https://github.com/godotengine/godot-proposals/issues/2425). 😢

Reply:
Yep, that's why I use this inside Ready until they implement it (without relying on 3rd party libraries), I also do this in Unity: initialize (cache) components once they enter the scene (Awake() in Unity or entering the Tree in Godot). This throws an exception if the component does not exist, so in my opinion this is a _good enough_ solution,  unless there is an actual proper way of doing this.